src/nvim/tui/tui.c

```diff
@@ -862,7 +862,7 @@ static void fix_terminfo(TUIData *data)
     data->unibi_ext.enter_replace_mode = (int)unibi_add_ext_str(ut, NULL,
                                                                 "\x1b[3 q");
     data->unibi_ext.exit_insert_mode = (int)unibi_add_ext_str(ut, NULL,
-                                                              "\x1b[2 q");
+                                                              "\x1b[1 q");
```

Comment:
Actually, it seems like it doesn't make a difference in vte3 0.42.5; both values behave correctly.